llama-index-integrations/readers/llama-index-readers-feishu-wiki/llama_index/readers/feishu_wiki/base.py

```diff
@@ -0,0 +1,150 @@
+
+        document_ids = self._load_space(space_id, parent_node_token=parent_node_token)
+        document_ids = list(set(document_ids))
+
+        results = []
+        for document_id in document_ids:
+            doc = self._load_doc(document_id)
+            results.append(Document(text=doc, extra_info={"document_id": document_id}))
+        return results
+
+    def _load_space(self, space_id, parent_node_token=None) -> str:
+        if self.tenant_access_token == "" or self.expire < time.time():
+            self._update_tenant_access_token()
+        headers = {
+            "Authorization": f"Bearer {self.tenant_access_token}",
+            "Content-Type": "application/json; charset=utf-8",
+        }
+
+        url = self.host + self.wiki_spaces_url_path.format(space_id)
+        if parent_node_token:
+            url += f"?parent_node_token={parent_node_token}"
+        try:
+            response = requests.get(url, headers=headers)
+            result = response.json()
+        except Exception:
+            return []
+        if not result.get("data"):
+            return []
+        obj_token_list = []
+        for item in result["data"]["items"]:
+            obj_token_list.append(item["obj_token"])
+            if item["has_child"]:
+                child_obj_token_list = self._load_space(
+                    space_id=space_id, parent_node_token=item["node_token"]
+                )
+                if child_obj_token_list:
+                    obj_token_list.extend(child_obj_token_list)
+        return obj_token_list
+
+    def _load_doc(self, document_id) -> str:
+        """Load a document from Feishu Docs.
+
+        Args:
+            document_id: the document id.
+
+        Returns:
+            The document text.
+        """
+        url = self.host + self.documents_raw_content_url_path.format(document_id)
+        if self.tenant_access_token == "" or self.expire < time.time():
+            self._update_tenant_access_token()
+        headers = {
+            "Authorization": f"Bearer {self.tenant_access_token}",
+            "Content-Type": "application/json; charset=utf-8",
+        }
+        try:
+            response = requests.get(url, headers=headers)
+            result = response.json()
+        except Exception:
+            return None
+        if not result.get("data"):
+            return None
+        return result["data"]["content"]
+
+    def _update_tenant_access_token(self):
+        """For update tenant_access_token."""
+        url = self.host + self.tenant_access_token_internal_url_path
+        headers = {"Content-Type": "application/json; charset=utf-8"}
+        data = {"app_id": self.app_id, "app_secret": self.app_secret}
+        response = requests.post(url, data=json.dumps(data), headers=headers)
+        self.tenant_access_token = response.json()["tenant_access_token"]
+        self.expire = time.time() + response.json()["expire"]
+
+    def set_lark_domain(self):
```

Comment:
name `set_lark_domain` suggests that method accepts value to set, but it doesn't

Reply:
Fixed it.